### PR TITLE
PA-4774 extended WhoamiV1 interface for Wonderwal

### PR DESCRIPTION
**Pull request by Luca-Esposito**

### CODE REVIEW

File: src/ps-web-apis.ts [hunk 1]
```diff
@@ -174,8 +174,31 @@ export interface WhoamiV1 {
...
+     *
+     * @param props - The props that should be passed, which will be used to render the Wonderwall.
+     * The props should contain the template and variant.
+     * The props can optional contain the headline and ctaText.
+     *
+     * @param callback - The callback that should be called after the main functionality is done.
+     * The callback should be a function
+     *
+     */
+    renderWonderwall(container: HTMLElement, props: WonderwallProps, callback?: () => void): void;
```
Comment (bwagneras) [inline src/ps-web-apis.ts:192]: shouldn't we give it a name which is more integrator friendly, like renderLogin?

**Review by bwagneras — APPROVED** [review]

File: src/ps-web-apis.ts [hunk 2]
```diff
@@ -174,8 +174,31 @@ export interface WhoamiV1 {
...
 }
 
+export type WonderwallProps = {
+    template: string;
```
Comment (fgerardwltn24) [inline src/ps-web-apis.ts:196]: What is the meaning of template / variant? , if login etc... let's fix this to "login" | "register",  ...

Comment (fgerardwltn24) [reply src/ps-web-apis.ts:196]: I wouldn't allow from the perspective of typing, plain strings.

File: src/ps-web-apis.ts (hunk 1 above)
Comment (fgerardwltn24) [reply src/ps-web-apis.ts:192]: renderAuthComponent(....) i guess we will use this method to display login or register

Comment (Luca-Esposito) [reply src/ps-web-apis.ts:192]: but verification and recovery is also possible. But i think for now we only will use login and register, right?

File: src/ps-web-apis.ts [hunk 3]
```diff
@@ -174,8 +174,31 @@ export interface WhoamiV1 {
...
+     *
+     * @param callback - The callback that should be called after the main functionality is done.
+     * The callback should be a function
+     *
+     */
+    renderAuthComponent(container: HTMLElement, props: WonderwallProps, callback: () => void): void;
 }
 
+export type WonderwallProps = {
+    template: "login" | "register" | "verification" | "recovery";
```
Comment (eschmittel) [inline src/ps-web-apis.ts:196]: Probably would only use login, register. I don't see use case here to expose them. I exposed them in the general web component as this is helpful for debugging the templates.

Comment (Luca-Esposito) [reply src/ps-web-apis.ts]: good point, i changed that

**Comment by Luca-Esposito** [comment]
we shouldn't go the way of a callback function, we should use Events.

https://stackoverflow.com/questions/50865614/can-i-pass-function-as-attribute-to-web-component

**Comment by eschmittel** [comment]
> we shouldn't go the way of a callback function, we should use Events.
> 
> for this reason: https://stackoverflow.com/questions/50865614/can-i-pass-function-as-attribute-to-web-component

Not necessarily for this function. The function itself could listen for the event when rednering the wonderwall and thus make it a bit easier for integrators.

**Review by eschmittel — APPROVED** [review]

**Pull request merged**